src/shell/shell_cmds.cpp

```diff
@@ -629,8 +629,9 @@ void DOS_Shell::CMD_DIR(char * args) {
 
 	if (!optB) {
 		if (print_label) {
-			const char *label = Drives[drive_idx]->GetLabel();
-			WriteOut(MSG_Get("SHELL_CMD_DIR_VOLUME"), drive_letter, label);
+			const auto label = To_Label(Drives[drive_idx]->GetLabel());
```

Comment:
Were there cases where `To_Label(..)` is needed to fixup the output from `GetLabel()`?  I did a bit of drilling into these call trees and see the cache and ISO drives both `Set_Label(..)` on the in-bound value; but I couldn't tell if `Set_Label(..)` extactly produces the cleanup that `Get_Label` performs.

More a curiosity. If `GetLabel()` still needs `To_Label()` for a "final cleanup", then maybe we should drop a **TODO:** on `GetLabel()` .. either rename it to something more intermediate or make it really return an end-state Label.

Reply:
I needed to add `To_Label` in here because otherwise label was displayed in 8.3 format instead of 11 characters. FreeDOS displays 11 characters in here, so that was my guiding principle.

I'm afraid changing the format returned by `GetLabel()` can have unintended consequences (the method seems to be used to fill-in internal DOS structures)… 

I added a TODO comment: 3e2da1ce7ef7923d17bb3d634d95e05e210f59d4